homeassistant/components/stream/const.py

```diff
@@ -15,8 +15,9 @@
 OUTPUT_IDLE_TIMEOUT = 300  # Idle timeout due to inactivity
 
 NUM_PLAYLIST_SEGMENTS = 3  # Number of segments to use in HLS playlist
-MAX_SEGMENTS = 4  # Max number of segments to keep around
+MAX_SEGMENTS = 5  # Max number of segments to keep around
```

Comment:
We increase the MAX_SEGMENTS by 1 because the last segment in the queue is usually incomplete. It gets pushed as soon as it is created, and we can access the last segment's parts as soon as they are added, but the whole segment is usually not available.

Reply:
Yes the effective number is still always 4 except for the brief period between when the last part gets pushed to the last segment and a new segment is created/pushed. It's essentially pushing an empty segment to the queue before filling it up as opposed to waiting until the segment is full before pushing it.
There should not be a change to latency in the non-ll case from this. There is essentially an increase of memory used on the segment being processed though - the `BytesIO` object is still around as it's getting filled up, and the `bytes` are getting copied out one chunk at a time, so when the segment is just finished there are 2 copies of the data in the segment instead of 1. I don't think this is that significant though (the increase from using longer segments would be a bigger issue if memory is a concern).